.github/workflows/conformance-aws-cni-v1.10.yaml

```diff
@@ -121,6 +122,12 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 45
     steps:
+      - name: Install pinned kubectl version
+        uses: azure/setup-kubectl@v2.0
+        id: kubectl-install
+        with:
+          version: ${{ env.kubectl_version }}
+
```

Comment:
Nit: we should avoid using third-party actions when possible, e.g. here we could pull `kubectl` manually:

```sh
curl -LO https://dl.k8s.io/release/${{ env.kubectl_version }}/bin/linux/amd64/kubectl && sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
```

Rationale: avoid the risk of third-party actions doing unwanted things under the hood.

Since this is a temporary solution and I think `azure` actions are trustable enough, it's OK, but I would say we should at least pin the action to a specific SHA to be consistent with how we do it everywhere else (to mitigate risk of pwn attacks).